lang/cs/Org.Apache.REEF.Tests/Functional/Bridge/TestSecurityToken.cs

```diff
@@ -0,0 +1,153 @@
+using Org.Apache.REEF.Utilities;
+using Org.Apache.REEF.Utilities.Logging;
+using Xunit;
+
+namespace Org.Apache.REEF.Tests.Functional.Bridge
+{
+    /// <summary>
+    /// Test set security tokens
+    /// </summary>
+    [Collection("FunctionalTests")]
+    public class TestSecurityToken
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(TestSecurityToken));
+
+        private const string DefaultPortRangeStart = "2000";
+        private const string DefaultPortRangeCount = "20";
+
+        private const string Identifier1 = "TrustedApplicationTokenIdentifier";
+        private const string Identifier2 = "TrustedApplicationTokenIdentifier2";
+        private const string TokenKey1 = "TrustedApplication007";
+        private const string TokenKey2 = "TrustedApplication008";
+        private const string Password1 = "none";
+        private const string Password2 = "none";
+
+        /// <summary>
+        /// This is to test pass tokens from client for Yarn application
+        /// </summary>
+        [Fact]
+        [Trait("Environment", "Yarn")]
+        [Trait("Priority", "1")]
+        [Trait("Description", "Run CLR Test on Yarn")]
+        public void TestSetMultipleSecurityToken()
+        {
+            IInjector injector = TangFactory.GetTang().NewInjector(GetRuntimeConfiguration());
+            var reefClient = injector.GetInstance<IREEFClient>();
+            var jobRequestBuilder = injector.GetInstance<JobRequestBuilder>();
+
+            var jobSubmission = jobRequestBuilder
+                .AddDriverConfiguration(GetDriverConfig())
+                .AddGlobalAssemblyForType(typeof(HelloDriver))
+                .SetJobIdentifier("MyTestJob")
+                .Build();
+
+            var result = reefClient.SubmitAndGetJobStatus(jobSubmission);
+            var state = PullFinalJobStatus(result);
+            Logger.Log(Level.Info, "Application final state : {0}.", state);
+
+            Assert.Equal(FinalState.SUCCEEDED, state);
+        }
+
+        /// <summary>
+        /// Use HelloDriver in the test
+        /// </summary>
+        /// <returns></returns>
+        private IConfiguration GetDriverConfig()
+        {
+            return DriverConfiguration.ConfigurationModule
+                .Set(DriverConfiguration.OnEvaluatorAllocated, GenericType<HelloDriver>.Class)
+                .Set(DriverConfiguration.OnDriverStarted, GenericType<HelloDriver>.Class)
+                .Build();
+        }
+
+        /// <summary>
+        /// Pull job final status until the Job is done
+        /// </summary>
+        /// <param name="jobSubmitionResult"></param>
+        /// <returns></returns>
+        private FinalState PullFinalJobStatus(IJobSubmissionResult jobSubmitionResult)
+        {
+            int n = 0;
+            var state = jobSubmitionResult.FinalState;
+            while (state.Equals(FinalState.UNDEFINED) && n++ < 100)
+            {
+                Thread.Sleep(3000);
```

Comment:
I am not sure we should do polling here. Can't we use `REEF.GetJobFinalStatus()` instead?

Reply:
Calling GetJobFinalStatus internally is the same thing. It returns the current value of the FinalStatus, it can be UNDEFINED if the job has not completed yet. Here I would like to see the status after the job is done.